torch/_dynamo/variables/tensor.py

```diff
@@ -424,6 +424,18 @@ def call_method(
         elif name == "__len__":
             return self.call_method(tx, "size", [ConstantVariable(0, **options)], {})
         elif name == "__setitem__":
+            if (
+                not (config.dynamic_shapes and config.capture_dynamic_output_shape_ops)
```

Comment:
This is sliding us a bit back on the dynamic shapes checks, but, since I am all in on FSDP atm, this is more than reasonable.

Do you mind just adding a comment as to why this is excluded like this?

Reply:
This is just following the same pattern we do for nonzero. When you kill config.dynamic_shapes you can just delete that from here and the knob is now just `not capture_dynamic_output_shape_ops`, which makes sense (because inductor can't deal with nonzero right now).